plugins/auto-sizes/hooks.php

```diff
@@ -90,3 +90,54 @@ function auto_sizes_render_generator(): void {
 	echo '<meta name="generator" content="auto-sizes ' . esc_attr( IMAGE_AUTO_SIZES_VERSION ) . '">' . "\n";
 }
 add_action( 'wp_head', 'auto_sizes_render_generator' );
+
+/**
+ * Filter the sizes attribute for images to improve the default calculation.
+ *
+ * @since n.e.x.t
+ *
+ * @param string               $content      The block content about to be rendered.
+ * @param array<string, mixed> $parsed_block The parsed block.
+ * @return string The updated block content.
+ */
+function auto_sizes_improve_image_sizes_attribute( string $content, array $parsed_block ): string {
+
+	$processor = new WP_HTML_Tag_Processor( $content );
+	$has_image = $processor->next_tag( array( 'tag_name' => 'img' ) );
+	$layout    = wp_get_global_settings( array( 'layout' ) );
+
+	// Only update the markup if an image is found.
+	if ( $has_image ) {
+
+		$align = $parsed_block['attrs']['align'] ?? null;
+		$sizes = null;
+		// Handle different alignment use cases.
+		switch ( $align ) {
+			case 'full':
+				$sizes = '100vw';
+				break;
+
+			case 'wide':
+				if ( isset( $layout['wideSize'] ) ) {
```

Comment:
Per https://github.com/WordPress/performance/issues/1219:

```suggestion
				if ( array_key_exists( 'wideSize', $layout ) ) {
```